tt_metal/hw/ckernels/wormhole_b0/metal/llk_io/llk_io_unpack.h

```diff
@@ -37,6 +37,27 @@ inline void llk_setup_operands() {
     }
 }
 
+// Apply delay on odd rows of tensix cores in case of matmul.
+// We do this so that not all cores start at once, therefore reducing the chance of di/dt problems.
+// MM_STAGGER_ODD_ROWS is an externally controlled define, set up in program compilation.
+inline __attribute__ ((__always_inline__)) void apply_mm_stagger(int operand) {
+    #ifdef MM_STAGGER_ODD_ROWS
+    static bool stagger_applied = false;
+    constexpr int stagger_operand = 1;
+    constexpr int stagger_delay_in_cycles = 12288;
+    if (stagger_applied == false && operand == stagger_operand) {
+        stagger_applied = true;
+        constexpr uint32_t noc_id = 0;
+        uint32_t noc_id_logical_reg = NOC_CFG_READ_REG(noc_id, NOC_ID_LOGICAL);
+        uint32_t my_logical_y = (noc_id_logical_reg >> NOC_ADDR_NODE_ID_BITS) & NOC_NODE_ID_MASK;
```

Comment:
data movement cores' fw defines my_x and my_y which would cover this, but we don't presently do this in trisc fw (we could).

Reply:
the code listed above runs on trisc0 so I added the code that identifies logical coordinate on trisc0 and based on that decides on the stagger.
is that what you had in mind?

Reply:
I am assuming you refer to this: https://github.com/tenstorrent/tt-metal/blob/main/tt_metal/hw/inc/risc_common.h#L134

I tried out this locally for triscs as well - I am just not sure though if it is ok to add it to trisc code in general since the stagger is applied only for matmuls on 48+ cores? We could also enable it for unpacker thread only as that is where it is used.